app/controllers/LongerCommodityCodeController.scala

```diff
@@ -52,31 +52,48 @@ class LongerCommodityCodeController @Inject() (
 
   private val form = formProvider()
 
-  def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
-      val preparedForm = request.userAnswers.get(LongerCommodityCodePage(recordId)) match {
-        case None        => form
-        case Some(value) => form.fill(value)
-      }
+  def onPageLoad(mode: Mode, recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    val categorisationInfoOpt = request.userAnswers.get(RecordCategorisationsQuery).flatMap(_.records.get(recordId))
 
-      val commodityCodeOption = request.userAnswers
-        .get(RecordCategorisationsQuery)
-        .flatMap(x => x.records.get(recordId).map(x => x.commodityCode))
+    val originalComcodeOpt = categorisationInfoOpt.map(_.originalCommodityCode).getOrElse(None)
```

Comment:
.getOrElse(None) - if it's an option it'll naturally do that

Reply:
i.e. it will only map if it's defined/not empty, otherwise it'll return None

Reply:
It's a nested option, I'll assess whether it needs to be one or not as it just seems weird to do this anyway

Reply:
Removed nested option and get or else :) Committing soon.